### DISCUSSION TITLE
Drawing different shapes with Imagemagick-Clip the figure to the boundary

### DISCUSSION THREAD

**Discussion by srini8080** [ImageMagick/ImageMagick#6262] [Digital Image Processing]
Linux
Imagemagick 7

[Drawing different shapes with Imagemagick](https://legacy.imagemagick.org/discourse-server/viewtopic.php?p=166605#p166605)

Fred gave a script here 

num=6
rad=150
diam=$((2*rad))
offset=30
ptArr=()
for ((i=0; i<num; i++)); do
ang=$((i*360/num + offset))
xx[$i]=`convert xc: -format "%[fx:$rad*cos($ang*pi/180)+$rad-0.5]" info:`
yy[$i]=`convert xc: -format "%[fx:$rad*sin($ang*pi/180)+$rad-0.5]" info:`
ptArr[$i]="${xx[$i]},${yy[$i]}"
done
convert -size ${diam}x${diam} xc:black -fill white -draw "polygon ${ptArr[*]}" -alpha off \
-trim +repage -bordercolor black -border 1 hexagon.gif

Can the script clip the image to the boundary of the polygon. Also when I give the sides as 4 or 5 (num=4 or num=5), the polygon is not sitting on its base.

Thanks,

**Comment by fmw42** [comment]
Change the offset angle to accommodate the number of sides and starting vertex.

As to the clipping, it has to do with the shape of the polygon w.r.t the size of the image. A hexagon will have slightly different width than height if I recall.  So you have to adjust ${diam}x${diam} for the shaped polygon and its orientation.  This is not trivial.

**Comment by snibgo** [comment]
> ... the polygon is not sitting on its base

Perhaps `-flip` will do what you want.

> Can the script clip the image to the boundary of the polygon.

I'm not sure what you want.

If you make a white polygon on a black background, you can "`-compose CopyOpacity -composite`" that to another image. The effect will be that image pixels outside the polygon will become transparent. Is that what you want?

**Reply by srini8080** [reply]
I am looking to create polygons that sit on their largest side, which I can put on a color background.

Thanks

**Reply by fmw42** [reply]
Sit on their largest side in width or in height or do you care?

**Reply by srini8080** [reply]
Nope. Either way is fine.

Thanks

**Comment by fmw42** [comment]
Does this do what you want? Should work for num=3, 4, 5, 6, 7 ...

```
num=6
rad=150
diam=$((2*rad))
offset=$((360/(2*$num)+90))
ptArr=()
for ((i=0; i<num; i++)); do
ang=$((i*360/num + offset))
xx[$i]=`convert xc: -format "%[fx:$rad*cos($ang*pi/180)+$rad-0.5]" info:`
yy[$i]=`convert xc: -format "%[fx:$rad*sin($ang*pi/180)+$rad-0.5]" info:`
ptArr[$i]="${xx[$i]},${yy[$i]}"
done
convert -size ${diam}x${diam} xc:black -fill white -draw "polygon ${ptArr[*]}" -alpha off \
-trim +repage -bordercolor black -border 1 hexagon.gif

```

![hexagon](https://user-images.githubusercontent.com/13458218/233227051-7db31bfa-2038-4b59-ace1-0225892b3150.gif)

If you want the sides exactly on the edges, remove -bordercolor black -border 1.

**Reply by srini8080** [reply]
Thanks a bunch

This is exactly what I wanted

Regards,
